src/Node.hpp

```diff
@@ -115,13 +118,15 @@ class Node
   arduino::_107_::opencyphal::ThreadsafeRingBuffer<std::tuple<uint32_t, size_t, std::array<uint8_t, 8>, CanardMicrosecond>> _canard_rx_queue;
   std::map<CanardPortID, RxTransferData> _rx_transfer_map;
   std::map<CanardPortID, CanardTransferID> _tx_transfer_map;
+  std::map<CanardPortID, impl::SubscriptionBase *> _msg_subscription_map;
```

Comment:
Shouldn't you use smart pointers? Pretty much all coding standards out there outright prohibit raw pointers, and for a good reason.

Reply:
You've seen the code ... I'm not sure how to reasonably use smart pointers here. An instance of `Subscription` is a `std::shared_ptr`, here I'm just holding a reference to its base class for invoking the onTransferReceived function ... which is cleaned-up on the `on_destruction_cb`. Not beautiful, but functional. Do you have a better suggestion?

Reply:
A simple `std::map<CanardPortID, std::shared_ptr<impl::SubscriptionBase>>` should work here.

Reply:
Fixed in 0e7896dd99d3ed48a4b6b46e9bf7a7c40829af9a.